test/integration2/transaction/set-deferrable-constraint.spec.js

```diff
@@ -0,0 +1,120 @@
+const { getAllDbs, getKnexForDb } = require('../util/knex-instance-provider');
+const { isMssql, isOracle, isPostgreSQL } = require('../../util/db-helpers');
+
+describe('Transaction', () => {
+  describe('setDeferrableConstraint', () => {
+    getAllDbs().forEach((db) => {
+      describe(db, () => {
+        let knex;
+        const tableName = 'deferrableTestTable';
+        const tableName1 = 'deferrableTestTable1';
+        before(() => {
+          knex = getKnexForDb(db);
+
+          if (isMssql(knex)) {
+            // Enable the snapshot isolation level required by certain transaction tests.
+            return knex.raw(
+              `ALTER DATABASE :db: SET ALLOW_SNAPSHOT_ISOLATION ON`,
+              { db: knex.context.client.config.connection.database }
+            );
+          }
+        });
+
+        after(() => {
+          return knex.destroy();
+        });
+
+        beforeEach(async () => {
+          await knex.schema.createTable(tableName, (table) => {
+            table.integer('id').primary().notNull();
+          });
+          await knex.schema.createTable(tableName1, (table) => {
+            table.integer('id').primary().notNull();
+          });
+        });
+
+        afterEach(async () => {
+          await knex.schema.dropTable(tableName);
+          await knex.schema.dropTable(tableName1);
+        });
+
+        it('deferrable initially immediate unique constraint all row are checked at end of update', async () => {
+          if (isPostgreSQL(knex) || isOracle(knex) || isOracle(knex)) {
```

Comment:
generally it's better to explicitly skip unsupported cases rather than execute empty test for them, that results in cleaner test suite output

Reply:
Got it